plugin/gcp/instance.py

```diff
@@ -0,0 +1,239 @@
+        Zone operation.
+
+        :return: REST response with operation responsible for the instance
+        creation process and its status
+        :raise: GCPError if there is any problem with startup script file:
+        e.g. the file is not under the given path or it has wrong permissions
+        """
+        self.logger.info('Create instance')
+        body = self.to_dict()
+
+        if self.startup_script is not None:
+            try:
+                with open(self.startup_script, 'r') as script:
+                    item = {
+                        'key': 'startup-script',
+                        'value': script.read()
+                    }
+                    body['metadata']['items'].append(item)
+            except IOError as e:
+                self.logger.error(str(e))
+                raise GCPError(str(e))
+
+        return self.compute.instances().insert(
+            project=self.project,
+            zone=self.zone,
+            body=body).execute()
+
+    @blocking(True)
+    def delete(self):
+        """
+        Delete GCP instance.
+        Zone operation.
+
+        :return: REST response with operation responsible for the instance
+        deletion process and its status
+        """
+        self.logger.info('Delete instance')
+        return self.compute.instances().delete(
+            project=self.project,
+            zone=self.zone,
+            instance=self.name).execute()
+
+    @blocking(True)
+    def set_tags(self, tags):
+        """
+        Set GCP instance tags.
+        Zone operation.
+
+        :return: REST response with operation responsible for the instance
+        tag setting process and its status
+        """
+        # each tag should be RFC1035 compliant
+        self.logger.info('Set tags')
+        self.tags.extend(tags)
+        self.tags = list(set(self.tags))
+        fingerprint = self.get()["tags"]["fingerprint"]
+        return self.compute.instances().setTags(
+            project=self.project,
+            zone=self.zone,
+            instance=self.name,
+            body={"items": self.tags, "fingerprint": fingerprint}).execute()
+
+    @blocking(True)
+    def remove_tags(self, tags):
+        """
+        Remove GCP instance tags.
+        Zone operation.
+
+        :return: REST response with operation responsible for the instance
+        tag removal process and its status
+        """
+        # each tag should be RFC1035 compliant
+        self.logger.info('Remove tags')
+        self.tags = [tag for tag in self.tags if tag not in tags]
+        fingerprint = self.get()["tags"]["fingerprint"]
+        return self.compute.instances().setTags(
+            project=self.project,
+            zone=self.zone,
+            instance=self.name,
+            body={"items": self.tags, "fingerprint": fingerprint}).execute()
+
+    def get(self):
+        """
+        Get GCP instance details.
+        """
+        self.logger.info("Get instance details")
```

Comment:
it would help if more details are printed to the log (all that we have - instance name, project and zone)